### DISCUSSION TITLE
How to resolve unhandle exception in a aws serverless application?

### DISCUSSION THREAD

**Discussion by Chanaka-Prasanna** [community/community#86610] [Programming Help]
### Select Topic Area

Bug

### Body

This is a sample serverless application I built using AWS serverless. It gives me an error mentioned at the end. Can anyone help me to resolve it? 

serverless.yml

`service: conversation-api
frameworkVersion: "3"
plugins:
  - serverless-offline

provider:
  name: aws
  runtime: nodejs18.x

functions:
  conversation-api:
    handler: index.handler
    events:
      - http:
          path: /
          method: get
          cors: true
      - http:
          path: /{proxy+}
          method: any
          cors: true
`
src/idex.html
`
const express = require("express");
const bodyParser = require("body-parser");
const cors = require("cors");
const routes = require("./ChatApp/routes.js");
const app = express();

app.use(bodyParser.json());
app.use(cors());

app.use("/chat-app", routes);

module.exports.app = app;`

index.html
`
const awsServerlesExpress = require("aws-serverless-express");

const app = require("./src/index.js");
try {
  const server = awsServerlesExpress.createServer(app);
  module.exports.handler = async (event, context) => {
    try {
      return awsServerlesExpress.proxy(server, event, context);
    } catch (error) {
      console.log(error);
    }
  };
} catch (error) {
  console.log(error);
}

crc/ChatApp/routes.js

const express = require("express");

const routes = express.Router({
  mergeParams: true,
});

routes.get("/", (req, res) => {
  res.status(200).json({ message: "Hello Chanaka" });
});

module.exports = routes;`

The error occured

> 
> × Uncaught exception
> Environment: win32, node 18.10.0, framework 3.38.0 (local) 3.38.0v (global), plugin 7.2.0, SDK 4.5.1
> Docs:        docs.serverless.com
> Support:     forum.serverless.com
> Bugs:        github.com/serverless/serverless/issues
> 
> Error:
> Error [ERR_UNHANDLED_ERROR]: Unhandled error. ({})
>     at new NodeError (node:internal/errors:393:5)
>     at Worker.emit (node:events:502:17)
>     at Worker.emit (node:domain:489:12)
>     at [kOnErrorMessage] (node:internal/worker:290:10)
>     at [kOnMessage] (node:internal/worker:301:37)
>     at MessagePort.<anonymous> (node:internal/worker:202:57)
>     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:731:20)
>     at exports.emitMessage (node:internal/per_context/messageport:23:28)
> PS E:\Projects\conversation-api> 
>

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Chanaka-Prasanna ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by DTS-11** [comment]
Hey there,

It looks like you're facing an unhandled error in your Serverless app on AWS. Have you tried checking your logs, debug the code, and ensure all dependencies are installed. Also try `console.log()` and try to find where the error is coming from. Look out for syntax issues and AWS Lambda handler and Serverless Express setup.

**Reply by Chanaka-Prasanna** [reply]
Sure DTS-11,I will check on your answer and let you know..
Thank you for your answer...

**Reply by DTS-11** [reply]
no problem! 😊

**Comment by Naimulnobel** [comment]
The error message "Unhandled error. ({})" is quite generic and doesn't provide specific details about what went wrong. However, there are a few things you can check and correct in your code:

1. **Typo in file name**: In your `serverless.yml` file, you have mentioned the handler file as `index.handler` but the file name you provided is `idex.html`. It should be `index.js`.

2. **AWS Lambda Runtime**: AWS Lambda currently supports Node.js 14.x as the latest version. You have mentioned `nodejs18.x` in your `serverless.yml` file, which is not supported. Change it to `nodejs14.x`.

3. **Serverless Offline Plugin**: You have mentioned `serverless-offline` in the plugins section of your `serverless.yml` file, but it's not properly formatted. It should be like this:

```yaml
plugins:
  - serverless-offline
```

4. **Express App Export**: In your `src/index.js` file, you are exporting `app` but in AWS Lambda, you should export a function. You have correctly created the server in `index.html` and exported the handler function. So, you might not need to export `app` in `src/index.js`.

5. **File Path**: In your `routes.js` file, the path seems to be `crc/ChatApp/routes.js`. If it's a typo, correct it to `src/ChatApp/routes.js`.

After making these changes, try to redeploy your Serverless application. If the error persists, you might want to add more logging to your code to get more details about what's going wrong.

**Reply by Chanaka-Prasanna** [reply]
Thank you for your reply Naimulnobel. I noticed what you mentioned here. Also, I found that there is no need to create a server using node js because this is a serverless application. The problems you mentioned and this problem may caused to the error.
